### PR TITLE
Bit simplification at System.Security.Cryptography.Xml

### PR DESCRIPTION
**Pull request by JaedsonBarbosa**
I just removed unnecessary itens.

### CODE REVIEW

File: src/Common/src/System/SR.cs [hunk 1]
```diff
@@ -16,7 +15,6 @@ private static ResourceManager ResourceManager
 
         // This method is used to decide if we need to append the exception message parameters to the message when calling SR.Format.
         // by default it returns false.
-        [MethodImpl(MethodImplOptions.NoInlining)]
```
Comment (stephentoub) [inline src/Common/src/System/SR.cs]: Why did you remove this?
cc: @tarekgh

File: src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.cs [hunk 2]
```diff
@@ -61,7 +63,7 @@ public sealed partial class EncryptedKey : System.Security.Cryptography.Xml.Encr
         public EncryptedKey() { }
         public string CarriedKeyName { get { throw null; } set { } }
         public string Recipient { get { throw null; } set { } }
-        public System.Security.Cryptography.Xml.ReferenceList ReferenceList { get { throw null; } }
+        public List<EncryptedReference> ReferenceList { get { throw null; } }
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.cs]: This is changing a public API.  This is not a compatible change.

File: src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.cs [hunk 3]
```diff
@@ -86,7 +88,7 @@ public abstract partial class EncryptedType
         public virtual System.Security.Cryptography.Xml.CipherData CipherData { get { throw null; } set { } }
         public virtual string Encoding { get { throw null; } set { } }
         public virtual System.Security.Cryptography.Xml.EncryptionMethod EncryptionMethod { get { throw null; } set { } }
-        public virtual System.Security.Cryptography.Xml.EncryptionPropertyCollection EncryptionProperties { get { throw null; } }
+        public virtual List<EncryptionProperty> EncryptionProperties { get { throw null; } }
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.cs]: Ditto

File: src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.cs [hunk 4]
```diff
@@ -160,34 +162,6 @@ public sealed partial class EncryptionProperty
         public System.Xml.XmlElement GetXml() { throw null; }
         public void LoadXml(System.Xml.XmlElement value) { }
     }
-    public sealed partial class EncryptionPropertyCollection : System.Collections.ICollection, System.Collections.IEnumerable, System.Collections.IList
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Xml/ref/System.Security.Cryptography.Xml.cs]: This is public API surface area.  It should not be removed.

**Comment by stephentoub** [comment]
@JaedsonBarbosa, thanks for the interest in contributing.  I started reviewing, but it looks like you're removing/changing public surface area.  That's generally not an acceptable change, as it breaks compatibility with the .NET Framework, existing code, etc.  We welcome PRs that clean up code and simplify it, but it needs to be compatible and shouldn't touch public surface area.

**Comment by danmosemsft** [comment]
We'd certainly welcome a separate change that removed/sorted usings, which I see you did a lot of.

File: src/Common/src/System/SR.cs (hunk 1 above)
Comment (tarekgh) [inline src/Common/src/System/SR.cs]: Please don't remove this. removing it will have the code optimized and we used to override this method in the .Net Native toolchain. 

---
In reply to: [111625461](https://github.com/dotnet/corefx/pull/18386#discussion_r111625461) [](ancestors = 111625461)

**Comment by karelz** [comment]
@JaedsonBarbosa if you're looking for contribution guidance, you can find it linked from our [main page](https://github.com/dotnet/corefx#contributing-guide).
As you can see it is now under construction, so if you have suggestion for improvements and changes, please let us know. Thanks!

**Comment by karelz** [comment]
Here's some specific [guidance](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/contributing.md).
Typically, it is expected that independent changes of different variety are submitted in separate PRs (e.g. removing dead code vs. adding new APIs vs. making code changes / minor fixes).
Please let us know if you have any questions, we will be happy to help.

**Comment by JaedsonBarbosa** [comment]
@karelz Ok 👌, so I will create other pull request with just removing dead code, and, after that, I will make other PR with small optimizations, like delete of internal or private unnecessary methods and classes.

**Comment by JaedsonBarbosa** [comment]
I was trying to make it work in my UWP project, but it's very more difficult than I thought and I don't know how can I do that 😞 
But I won't give up and first I will optimize and simplify the code, after that I will try to use that in my UWP project because I can't keep waiting.

**Comment by JaedsonBarbosa** [comment]
And do you know how can I undo every changes I made in my fork of this repository?

**Comment by danmosemsft** [comment]
> undo every changes I made in my fork of this repository?

If you just want to abandon this PR, just close it (I'll do htat now)
 If you want to tidy up by deleting the branch do `git push origin --delete <branch_name>`

**Comment by JaedsonBarbosa** [comment]
Thanks 😉

**Comment by karelz** [comment]
> I was trying to make it work in my UWP project, but it's very more difficult than I thought and I don't know how can I do that

That's what I expected. By trying to port it into UWP (i.e. .NET Core 1.1), you might easily need half of the 2.0 work in 1.1 as well. And that's a LOT of work.

> But I won't give up and first I will optimize and simplify the code, after that I will try to use that in my UWP project because I can't keep waiting.

I wish you good luck. I just want to emphasize that we will likely not take such changes into CoreFX code base, but it might unblock you for the next ~2 months.

**Comment by JaedsonBarbosa** [comment]
OK, I think I will need at least one or two weeks to finish that 🤔😓

**Comment by JaedsonBarbosa** [comment]
@karelz It now compile in a UWP class library 🎆, you can see that at https://github.com/JaedsonBarbosa/corefx/tree/BigOptimization?files=1
I'll now verify if it works well in my UWP app 😃

**Comment by JaedsonBarbosa** [comment]
The Project is in https://github.com/JaedsonBarbosa/corefx/tree/BigOptimization/src/System.Security.Cryptography.Xml/UWP

**Comment by karelz** [comment]
Very cool! Let us know if it does work ... others will may like to get their hands on your solution!

**Comment by JaedsonBarbosa** [comment]
But it is a very light solution.
